internal/updater/updater.go

```diff
@@ -64,7 +64,7 @@ func (u *Updater) Apply(ctx context.Context, b *rukpakv1alpha1.Bundle) error {
 	})
 }
 
-func EnsureCondition(condition metav1.Condition) UpdateStatusFunc {
+func EnsureCondition(condition metav1.Condition) UpdateBundleStatusFunc {
```

Comment:
Is it fair for `BundleStatus` to get to use this function name? :)

In all seriousness, I wonder if we should have `updater/bundle` and `updater/bundleinstance` subpackages such that we don't have to worry about de-conflicting names like this? Or maybe just demote this to `EnsureBundleCondition`? I worry about the length of the function names with the latter option though.

Reply:
Yes -- I agree that the naming conventions currently proposed in this PR add a bit of complexity. I can move the updaters into their own subpackages and import them, which hopefully makes things more clear and avoids collisions.